crawl4ai/adaptive_crawler.py

```
@@ -0,0 +1,1861 @@
+            result = await self.crawler.arun(url=url, config=config)
+            # Extract the actual CrawlResult from the container
+            if hasattr(result, '_results') and result._results:
+                result = result._results[0]
+
+            # Filter our all links do not have head_date
+            if hasattr(result, 'links') and result.links:
+                result.links['internal'] = [link for link in result.links['internal'] if link.get('head_data')]
+                # For now let's ignore external links without head_data
+                # result.links['external'] = [link for link in result.links['external'] if link.get('head_data')]
+
+            return result
```

Ruff (0.14.0)

Consider moving this statement to an `else` block

(TRY300)